### PR TITLE
Command Objects

### PR DESCRIPTION
**Pull request by stgeneral**

### CODE REVIEW

File: guides/command_objects.md [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+
+  def errors
+    @_errors ||= ActiveModel::Errors.new(self)
+  end
+
+  def fail!
+    @_success = false
+  end
+
+  def validate
```
Comment (ColinMcRae) [inline guides/command_objects.md:43]: What kind of validations will be here?

Comment (stgeneral) [reply guides/command_objects.md:43]: Anything that may or should be validated before command execution. But I haven't really played with use cases for validations yet. I've just outlined some methods like `validate`, `authorize` that might be useful.

Any thoughts are welcomed :)

Comment (ColinMcRae) [reply guides/command_objects.md:43]: Well, i used separated validator objects. And it looked like this:
```
user = User.new(user_params)
user_validator = UserRegistrationValidator.new(user)
if user_validator.valid?
  UserRegistrationCommand.call(user)
else
  user_validator.errors
  #errors processing
end
```

So, we will use command for business logic processing, and validators for data validations.
Also, im not sure, if we need to place authorize tools inside, but they use sometimes inside commands

Comment (ColinMcRae) [reply guides/command_objects.md:43]: https://api.rubyonrails.org/classes/ActiveModel/Validator.html

Comment (stgeneral) [reply guides/command_objects.md:43]: Seems that the purpose of `UserRegistrationValidator` is very close to [Form Objects](https://github.com/ergoserv/handbook/blob/master/guides/form_objects.md) we already used. The benefits of the `Form Object` approach is that it provides the object that can be nicely rendered on the view. 

As for the authorization tools. Sometimes commands may require to check the policy of the user, but probably you are right. We can remove this from basic command add some conventions or move authorization tools to a module, which could be included when needed.

**Comment by stgeneral** [comment]
Replaced with #7.

**Pull request closed without merging**